### DISCUSSION TITLE
Docker dependency not detected in "COPY --from" if used in combination with "--link"

### DISCUSSION THREAD

**Discussion by klaasdellschaft** [renovatebot/renovate#29654] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Platform: GitLab, Version: 37.407.1

### Please tell us more about your question or problem

Renovate is not discovering Docker dependencies if they are used in a `COPY --from=some-image:1.0.0` if used in conjunction with the `--link` option in the `COPY` statement. The behavior of Renovate depends on the order in which `--from` and `--link` are defined.

Example:

```dockerfile
FROM debian:bookworm-slim
COPY --link --from=hashicorp/terraform:1.8.5 /bin/terraform /usr/local/bin/terraform
COPY --from=alpine/helm:3.14.4 --link /usr/bin/helm /usr/local/bin/helm
```

In this example, only the `alpine/helm:3.14.4` dependency is discovered by Renovate while the `hashicorp/terraform:1.8.5` is not recognized.

The problem seems to be caused by the RegEx for discovering the `COPY --from` dependencies: https://github.com/renovatebot/renovate/blob/7178da30743b88b289d2f2cd859891d4c07be3ab/lib/modules/manager/dockerfile/extract.ts#L384

```
const copyFromRegex = new RegExp(
      '^[ \\t]*COPY(?:' +
        escapeChar +
        '[ \\t]*\\r?\\n| |\\t|#.*?\\r?\\n|--[a-z]+=[a-zA-Z0-9_.:-]+?)+--from=(?<image>\\S+)',
      'im',
    );
```

more specifically by the `--[a-z]+=[a-zA-Z0-9_.:-]+?` sub-expression. It assumes that all `--` options are always followed by `=some-value`.

**Comment by viceice** [comment]
clearly a bug, feel free to send a fix PR 🤗
